src/useCases/ReportManager.java

```diff
@@ -62,4 +64,12 @@ public ArrayList<ReportData> getReportData(PatientData patientData) {
                 filter(p -> p.getPatientId().equals(patientData.getId())).map(ReportData::new));
     }
 
+    public DoctorData getReportDoctor(ReportData report) {
+        Doctor doctor = doctorDatabase.get(report.getDoctorId());
+        if (doctor == null) {
+            return null;
+        }
+        return new DoctorData(doctor);
+    }
+
```

Comment:
This method should probably exist in DoctorManager, where it takes in a doctor id rather than a reportData. When called in the controller, simply extract the id from the report data.

Reply:
You are correct!